### ISSUE TITLE
The best long form writing experience on a smartphone

### ISSUE THREAD

**Issue by sawtdakhili** [open-source-ideas/ideas#363]
Before we start, let’s be clear. I am not a developer. I am not selling anything. This is merely just an idea based on my own insights as a creative writer. I would love to have different opinions on this one.

**Problem:**
It’s really inconvenient to write long-form content on a smartphone. When the phone is in portrait mode, there’s enough room to see all the text, but the keyboard is too tiny for extended writing sessions resulting in fatigue. When you turn the smartphone to landscape mode, two problems arise: the keyboard takes up two-thirds of the screen, making it hard to see what you’re typing, and reaching the keys in the middle of the keyboard becomes difficult. Thumb mode isn’t a satisfactory solution either, as it leaves a sad stupid wasted space in the middle of the split keyboard.

When writing on a smartphone, screen real estate is precious. Tablets offer a split thumb floating keyboard mode, but they are bulky, heavy, and not comfortable for long-term typing. Carrying a tablet everywhere isn’t practical.

**What’s available:**
I’ve tried a dozen writing apps (even Joe’s Termux with its high-condensed text) combined with several Android keyboards, but none come close to the experience of having a real keyboard (I prefer ortholinear split keyboards btw) attached to a smartphone for travel.

So, how can we fix the typing experience on a phone?

**Solution:**
The ideal solution would be to have a split keyboard that occupies the two top sides of the screen, with the text editor displayed in the middle. Full-screen mode to get rid of the status bar would be a plus. You would hold your phone like a Steam Deck or, even better, like the Nokia N-Gage (I know that there have been phones closer to what I’m describing, but the N-Gage is the most well-known form factor).

**Additional benefits:**
This layout would be more comfortable to hold, similar to how you hold your phone when playing PUBG. Your index fingers would rest on the phone, and the bottom corners of the phone would fit into the palms’ cavity. Achieving this would be possible since the keyboard would be positioned closer toward the top corners of the phone (remember, the phone is in landscape mode).

**Some other thoughts:**
Perhaps combining a keyboard with a text editor is excessive. It’s possible that a new type of keyboard that splits and allows you to view the text you’re typing in the middle would suffice. However, it’s uncertain whether this would work with every app.

**Disclaimer:**
#Please note that the following mockups provided are intended solely for illustrative purposes to convey the concept and potential user experience.

![termux-mockup](https://github.com/open-source-ideas/ideas/assets/70063536/dc026d67-589a-4125-a419-3e576cf170aa)
![termux](https://github.com/open-source-ideas/ideas/assets/70063536/81f6286d-019f-43f1-9aad-ba7a73cfabf0)
![long-form-writing-layout](https://github.com/open-source-ideas/ideas/assets/70063536/928435d0-66e3-4a6d-9d74-1c7da903d4d8)
![obsidian-layout](https://github.com/open-source-ideas/ideas/assets/70063536/dcf1a80e-cfae-4bd7-b604-258bcf1d5fae)
![apple-mail-layout](https://github.com/open-source-ideas/ideas/assets/70063536/a23b4e12-c333-4bca-bd50-b6a04c67cc8b)

**Comment by FredrikAugust** [comment]
Could you please add back the tags?
Thanks.

**Comment by sawtdakhili** [comment]
> Could you please add back the tags? Thanks.

I don't know how ^^'

**Comment by tyrianist** [comment]
It is a good idea. As a writer, do you have ideas for other features, in addition to the keyboard, for the text editor?

**Comment by FredrikAugust** [comment]
@sawtdakhili 

When you create an idea, there's a list of checkboxes you can tick. Please create a new issue and copy in the relevant checkboxes. People use it to get a quick understanding of the scope and complexity of an idea.

**Comment by sawtdakhili** [comment]
I made a new posting with proper tags.
